naxsi_src/naxsi_skeleton.c

```diff
@@ -1159,3 +1272,473 @@ static ngx_int_t ngx_http_dummy_access_handler(ngx_http_request_t *r)
   return NGX_DECLINED;
 }
 
+
+static ngx_int_t
+ngx_http_naxsi_add_variables(ngx_conf_t *cf)
+{
+    ngx_http_variable_t *var, *v;
+
+    for (v = ngx_http_naxsi_variables; v->name.len; v++) {
+        var = ngx_http_add_variable(cf, &v->name, v->flags);
+        if (var == NULL) {
+            return NGX_ERROR;
+        }
+        var->get_handler = v->get_handler;
+        var->data = v->data;
+    }
+    return NGX_OK;
+}
+
+
+static ngx_http_request_ctx_t*
+recover_request_ctx(ngx_http_request_t *r) {
+    ngx_http_request_ctx_t *ctx;
+    ngx_pool_cleanup_t *cln;
+    ctx = ngx_http_get_module_ctx(r, ngx_http_naxsi_module);
+    if (ctx == NULL && (r->internal || r->filter_finalize)) {
+        for (cln = r->pool->cleanup; cln; cln = cln->next) {
+            if (cln->handler == ngx_http_dummy_cleanup_handler) {
+                ctx = cln->data;
+                break;
+            }
+        }
+    }
+    return ctx;
+}
+
+
+static ngx_int_t
+ngx_http_naxsi_server_variable(ngx_http_request_t *r, ngx_http_variable_value_t *v, uintptr_t data)
+{
+    v->data = r->headers_in.server.data;
+    v->len = r->headers_in.server.len;
+    v->valid = 1;
+    v->no_cacheable = 0;
+    v->not_found = 0;
+    return NGX_OK;
+}
+
+
+static ngx_int_t
+ngx_http_naxsi_uri_variable(ngx_http_request_t *r, ngx_http_variable_value_t *v, uintptr_t data)
+{
+    ngx_http_request_ctx_t *ctx = recover_request_ctx(r);
+    if (!ctx) {
+        v->not_found = 1;
+        return NGX_OK;
+    }
+
+    ngx_str_t *tmp_uri = ngx_pcalloc(r->pool, sizeof(ngx_str_t));
+    if (!tmp_uri) {
+        return (NGX_ERROR);
+    }
+    tmp_uri->len = r->uri.len + (2 * ngx_escape_uri(NULL, r->uri.data, r->uri.len, NGX_ESCAPE_ARGS));
+    tmp_uri->data = ngx_pcalloc(r->pool, tmp_uri->len+1);
```

Comment:
Even if this seems a bit exotic, this is actually a good quesiton, as it seems user can change the max uri len accepted :
https://stackoverflow.com/questions/1067334/how-to-set-the-allowed-url-length-for-a-nginx-request-error-code-414-uri-too